File: HappyNumber/happy_number.py
# https://leetcode.com/problems/happy-number/

class Solution:
    def happy_number(self, number):
        original_number = number
        while number != 1:
            number = sum([int(char) ** 2 for char in str(number)])
            if number == original_number:
                return False
        return True

if __name__ == '__main__':
    solution = Solution()
    print(solution.happy_number(20))

File: HappyNumber/happyNumber.js
// https://leetcode.com/problems/happy-number/

function happyNumber(number) {
    let original_number = number;
    while (number !== 1) {
        let stringNumber = number.toString();
        number = 0;
        for (char of stringNumber) {
            number += parseInt(char) ** 2;
        }
        if (number == original_number) {
            return false;
        }
    }
    return true;
}

console.log(happyNumber(20));
